pkg/metadata/metadata.go

```diff
@@ -27,10 +29,36 @@ type metadataServiceManager struct {
 
 var _ MetadataService = &metadataServiceManager{}
 
-func NewMetadataService(linodeClient linodeclient.LinodeClient, host string) (metadata MetadataService, err error) {
-	linode, err := getLinodeByName(linodeClient, host)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't initialize Linode client: %s", err)
+// NewMetadataService retrieves information about the linode where the
+// application is currently running on.
+func NewMetadataService(linodeClient linodeclient.LinodeClient, nodeName string) (metadata MetadataService, err error) {
+	// Get information about the Linode this pod is executing in.
+	// Assume that the Linode ID file does not exist.
+	linodeInfo := nodeName
+	isID := false
+
+	// Attempt to get the Linode ID information
+	data, err := ioutil.ReadFile("/linode-info/linode-id")
+	if err == nil {
+		// File read was successful, use Linode ID to create metadata service
+		linodeInfo = string(data)
+		isID = true
+	}
+
+	// Linode instance
+	var linode *linodego.Instance
+	if isID {
+		// Search for Linode instance by ID
+		linode, err = getLinodeByID(linodeClient, linodeInfo)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to find Linode with ID %s: %v", linodeInfo, err)
```

Comment:
You can simply return `nil, err` here, because you have already created this Error struct.